resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala

```diff
@@ -99,9 +100,12 @@ private[spark] trait BasicTestsSuite { k8sSuite: KubernetesSuite =>
   }
 
   test("Run SparkRemoteFileTest using a remote data file", k8sTestTag) {
-    sparkAppConf
-      .set("spark.files", REMOTE_PAGE_RANK_DATA_FILE)
-    runSparkRemoteCheckAndVerifyCompletion(appArgs = Array(REMOTE_PAGE_RANK_FILE_NAME))
+    assert(sys.props.contains("spark.test.home"), "spark.test.home is not set!")
```

Comment:
Is this required in tests, @attilapiros ?
I'm wondering when does this happen because we set this explicitly on both Maven/SBT test environment setting.

Reply:
I followed the existing pattern:
```
$ rg "spark.test.home"
sql/hive/src/test/scala/org/apache/spark/sql/hive/SparkSubmitTestUtils.scala
46:      sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!")))

sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/SQLHelper.scala
90:    if (!(sys.props.contains("spark.test.home") || sys.env.contains("SPARK_HOME"))) {
91:      fail("spark.test.home or SPARK_HOME is not set.")
93:    sys.props.getOrElse("spark.test.home", sys.env("SPARK_HOME"))

project/SparkBuild.scala
1058:    javaOptions in Test += "-Dspark.test.home=" + sparkHome,

pom.xml
262:    <spark.test.home>${session.executionRootDirectory}</spark.test.home>
2654:              <spark.test.home>${spark.test.home}</spark.test.home>
2706:              <spark.test.home>${spark.test.home}</spark.test.home>

python/pyspark/tests/test_context.py
306:        conf = SparkConf().set("spark.test.home", SPARK_HOME)

python/pyspark/tests/test_taskcontext.py
309:        conf = SparkConf().set("spark.test.home", SPARK_HOME)

core/src/main/scala/org/apache/spark/api/r/RUtils.scala
36:    val sparkHome = sys.env.get("SPARK_HOME").orElse(sys.props.get("spark.test.home"))

core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala
148:      assert(sys.props.contains("spark.test.home"), "spark.test.home is not set!")
149:      new File(sys.props("spark.test.home"))

core/src/test/scala/org/apache/spark/deploy/worker/CommandUtilsSuite.scala
32:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))

core/src/test/scala/org/apache/spark/deploy/worker/ExecutorRunnerTest.scala
29:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))

core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala
728:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))
748:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))
1537:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))

core/src/test/scala/org/apache/spark/DriverSuite.scala
34:    val sparkHome = sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))

core/src/test/scala/org/apache/spark/util/SparkUncaughtExceptionHandlerSuite.scala
29:    sys.props.getOrElse("spark.test.home", fail("spark.test.home is not set!"))

core/src/test/scala/org/apache/spark/launcher/LauncherBackendSuite.scala
50:      .setSparkHome(sys.props("spark.test.home"))

resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala
150:      // Try the spark test home
151:      sys.props("spark.test.home")

resource-managers/yarn/src/test/scala/org/apache/spark/launcher/TestClasspathBuilder.scala
28:  childEnv.put(CommandBuilderUtils.ENV_SPARK_HOME, sys.props("spark.test.home"))

resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala
207:      .setSparkHome(sys.props("spark.test.home"))
310:    val sparkHome = sys.props("spark.test.home")

resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/BaseYarnClusterSuite.scala
158:    launcher.setSparkHome(sys.props("spark.test.home"))

launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java
290:      System.getProperty("spark.test.home"));
308:      launcher.childEnv.put("SPARK_CONF_DIR", System.getProperty("spark.test.home")
415:    builder.childEnv.put(CommandBuilderUtils.ENV_SPARK_HOME, System.getProperty("spark.test.home"));

launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java
250:      path = System.getProperty("spark.test.home");
```